File: src/main/java/dev/retrotv/crypt/twe/des/DESCBC.java
package dev.retrotv.crypt.twe.des;

import java.security.Key;
import java.security.NoSuchAlgorithmException;

import javax.crypto.KeyGenerator;
import javax.crypto.spec.IvParameterSpec;

import dev.retrotv.crypt.exception.KeyGenerateException;
import dev.retrotv.crypt.twe.ParameterSpecGenerator;
import dev.retrotv.enums.Algorithm;
import dev.retrotv.utils.SecureRandomUtil;

public class DESCBC extends DES implements ParameterSpecGenerator<IvParameterSpec> {
    public DESCBC() {
        this.algorithm = Algorithm.DESCBC_PADDING;
    }
    
    @Override
    public Key generateKey() throws KeyGenerateException {
        try {
            KeyGenerator keyGenerator = KeyGenerator.getInstance("DES");
            return keyGenerator.generateKey();
        } catch (NoSuchAlgorithmException e) {
            throw new KeyGenerateException("NoSuchAlgorithmException: \n지원하지 않는 암호화 알고리즘 입니다.");
        }
    }

    @Override
    public IvParameterSpec generateSpec() {
        return new IvParameterSpec(SecureRandomUtil.generate(8));
    }
}

File: src/main/kotlin/dev/retrotv/crypto/twe/des/DESCBC.kt
package dev.retrotv.crypto.twe.des

import dev.retrotv.crypto.exception.KeyGenerateException
import dev.retrotv.crypto.twe.ParameterSpecGenerator
import dev.retrotv.enums.Algorithm
import dev.retrotv.utils.generate
import java.security.Key
import java.security.NoSuchAlgorithmException
import javax.crypto.KeyGenerator
import javax.crypto.spec.IvParameterSpec

/**
 * DES/CBC 양방향 암호화 클래스 입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
@Deprecated("해킹에 취약한 양방향 암호화 알고리즘 입니다.")
class DESCBC : DES(), ParameterSpecGenerator<IvParameterSpec> {

    init {
        algorithm = Algorithm.Cipher.DESCBC
    }

    override fun generateSpec(): IvParameterSpec {
        return IvParameterSpec(generate(8))
    }
}
